### PR TITLE
Feat service dataset

### PR DESCRIPTION
**Pull request by lurosenb**

### CODE REVIEW

File: service/dataset.py [hunk 1]
```diff
@@ -1,27 +1,126 @@
...
+
+#%%
+def register(dset):
+    dataset_name = dset["dataset_name"]
+
+    if dataset_name in DATASETS:
+        abort(401, "Dataset id {} already exists. Identifies must be unique".format(dataset_name))
+    
+    # Well-formed dset, post checks
+    wf_dset = {}
```
Comment (eedeleon) [inline service/dataset.py]: It is a service call so it is ok to work off of the dset, no need to replicate it

File: service/dataset.py [hunk 2]
```diff
@@ -1,27 +1,126 @@
...
+
+    # Add key if possible
+    if dset["key"] is "csv_details" or "dataverse_details":
+        wf_dset["key"] = dset["key"]
+    else:
```
Comment (eedeleon) [inline service/dataset.py]: if dset["key"] not in KNOWN_DATASET_KEYS: 
abort mission

File: sdk/opendp/whitenoise/client/__init__.py [hunk 3]
```diff
@@ -31,12 +32,26 @@ def submit(self, params, uri):
         return client.executerun(details)
 
 class DatasetClient(object):
+    def register(self, dataset):
+        client = _get_client()
+
+        if not 'csv_details' in dataset:
```
Comment (eedeleon) [inline sdk/opendp/whitenoise/client/__init__.py]: for key in known keys, if not avail set to None

File: service/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+## Swagger+autorest steps:
+1. Follow the steps in the swagger readme (under openapi/readme.md) and ensure that you can regenerate the restclient from the swagger.yml file using the autorest npm package. Regenerating it once ensures that any future errors are your fault : )
+2. Add a new path to the openapi/swagger.yml file. You can use one of the existing paths as a template, but make sure to modify each field and specify the api functionality carefully. Refer to online documentation for examples (https://swagger.io/docs/specification/describing-request-body/ is a good place to start) 
+3. As you define your new path, make sure to add schema definition that fits your specific use case under "definitions:" in the swagger.yml file. For example, /register takes in a very specific schema, which is defined in "DatasetPutDocument".
+4. Regenerate the swagger, and verify that the sdk/burdock/client/restclient contains your new path definition and sdk/burdock/client/restclient/models contains your new schema definition. (the names should be reflective of specifications from the swagger.yml file)
```
Comment (eedeleon) [inline service/README.md]: sdk/opendp/whitenoise/client/restclient

File: service/dataset.py [hunk 5]
```diff
@@ -1,34 +1,109 @@
...
+        abort(401, "Dataset id {} already exists. Identifies must be unique".format(dataset_name))
+
+    # Add key if possible
+    if dataset["dataset_type"] not in KNOWN_DATASET_TYPE_KEYS:
+        abort(402, "Given type was {}, must be either csv_details or dataverse_details.".format(str(dataset["dataset_type"])))
+    
+    # Add budget if possible 
+    if dataset["budget"]:
+        b = dataset["budget"]
+        if b <= 0.0: abort(403, "Budget must be greater than 0.")
```
Comment (eedeleon) [inline service/dataset.py]: we don't usually use multi line statements. Not sure why this b = dict lookup. dict lookup = b thing is happening

File: tests/service/datasets/test_data/example.csv [hunk 6]
```diff
@@ -0,0 +1,4 @@
+a,b,c,d
```
Comment (eedeleon) [inline tests/service/datasets/test_data/example.csv]: why is this copied?

File: service/dataset.py (hunk 5 above)
Comment (lurosenb) [reply service/dataset.py]: lol whoops

**Comment by lurosenb** [comment]
Cleaned up the comments Eddie had, should be pretty presentable. Note: dataverse smoke test currently failing due to no token for remote dset, and execute smoke tests failing after budget tracking. will investigate further.

**Pull request merged**